Bottom: 7cce5cae5d42cd31d7c3833ac60d0668c0c1d6d5
Top:    d375a8c3b149a38d9c0e691a32801dcd2977c4e9
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-07 14:03:51 -0700

Refresh of dbg-out

---

diff --git a/src/preload.c b/src/preload.c
index d1b6cff..4ebcc85 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -85,7 +85,7 @@ struct socket_calls {
 			  void *optval, socklen_t *optlen);
 	int (*fcntl)(int socket, int cmd, ... /* arg */);
 	int (*dup2)(int oldfd, int newfd);
-	int (*fstat)(int fd, struct stat *buf);
+	int (*fxstat64)(int ver, int fd, struct stat64 *buf);
 };
 
 static struct socket_calls real;
@@ -263,7 +263,7 @@ static void init_preload(void)
 	real.getsockopt = dlsym(RTLD_NEXT, "getsockopt");
 	real.fcntl = dlsym(RTLD_NEXT, "fcntl");
 	real.dup2 = dlsym(RTLD_NEXT, "dup2");
-	real.fstat = dlsym(RTLD_NEXT, "fstat");
+	real.fxstat64 = dlsym(RTLD_NEXT, "__fxstat64");
 
 	rs.socket = dlsym(RTLD_DEFAULT, "rsocket");
 	rs.bind = dlsym(RTLD_DEFAULT, "rbind");
@@ -968,20 +968,20 @@ int dup2(int oldfd, int newfd)
 	return newfd;
 }
 
-int __fxstat64(int ver, int socket, struct stat *buf)
+int __fxstat64(int ver, int socket, struct stat64 *buf)
 {
 	int fd, ret;
 
 	init_preload();
-	fprintf(fout, "fstat socket %d\n", socket);
+	fprintf(fout, "fxstat64 socket %d\n", socket);
 	if (fd_get(socket, &fd) == fd_rsocket) {
-		ret = real.fstat(socket, buf);
-		fprintf(fout, "fstat - rsocket %d\n", ret);
+		ret = real.fxstat64(ver, socket, buf);
+		fprintf(fout, "fxstat64 - rsocket %d\n", ret);
 		if (!ret)
 			buf->st_mode = (buf->st_mode & ~S_IFMT) | __S_IFSOCK;
 	} else {
-		ret = real.fstat(fd, buf);
-		fprintf(fout, "fstat - normal %d\n", ret);
+		ret = real.fxstat64(ver, fd, buf);
+		fprintf(fout, "fxstat64 - normal %d\n", ret);
 	}
 	fflush(fout);
 	return ret;
